Libraries/PushNotificationIOS/RCTPushNotificationManager.m

```diff
@@ -117,6 +139,33 @@ - (void)handleRemoteNotificationsRegistered:(NSNotification *)notification
                                               body:notification.userInfo];
 }
 
+- (void)handleLocalNotificationReceived:(NSNotification *)notification
+{
+  UILocalNotification *local = [notification.userInfo objectForKey:@"notification"];
+  NSArray *keys = [_scheduledLocalNotifications allKeysForObject:local];
+  NSString *identifier = [keys objectAtIndex:0];
+
+  [_scheduledLocalNotifications removeObjectForKey:identifier];
+
+  NSDictionary *userInfo = @{@"identifier": identifier};
+  [_bridge.eventDispatcher sendDeviceEventWithName:@"localNotificationReceived"
+                                              body:userInfo];
+}
+
+/**
+ * Remove all expired UILocalNotification instances that may still remain in _scheduledLocalNotifications
+ * since we can only explicity remove them if the app is active or the user taps the notification icon.
+ */
+- (void)handleApplicationDidBecomeActiveNotification:(__unused NSNotification *)notification
+{
+  for (NSString *identifier in [_scheduledLocalNotifications allKeys]) {
```

Comment:
nit: prefer dot syntax for properties, e.g. `_scheduledLocalNotifications.allKeys`, `note.fireDate.timeIntervalSinceReferenceDate`, etc.